src/include/miopen/batchnorm/common_spatial.hpp

```diff
@@ -268,15 +274,18 @@ inline bool UseMultiple(const miopen::batchnorm::ProblemDescription& problem)
     if(!problem.IsLayoutNHWC() &&
        problem.GetDirection() == miopen::batchnorm::Direction::Backward &&
        (!((in_nhw >= static_cast<size_t>(32 * 1024 * 1024) || in_cstride <= 1024) &&
+          (in_nhw >= static_cast<size_t>(32 * 1024 * 1024) || in_cstride <= 512) &&
           in_cstride > 512)))
```

Comment:
This looks overcomplicated, I think the truth table for this is the same as:
(!( (in_nhw >= static_cast<size_t>(32 * 1024 * 1024)) ))